File: lua/Java/LuaParserBase.java
import java.util.List;
import org.antlr.v4.runtime.*;
import java.io.PrintStream;
import java.util.HashSet;
import java.util.Set;

public abstract class LuaParserBase extends Parser
{
    protected LuaParserBase(TokenStream input) {
        super(input);
    }

    protected boolean IsFunctionCall()
    {
        BufferedTokenStream stream = (BufferedTokenStream)_input;
        var la = stream.LT(1);
	if (la.getType() != LuaLexer.NAME) return false;
	la = stream.LT(2);
	if (la.getType() == LuaLexer.OP) return false;
	return true;
    }
}

File: lua/Python3/LuaParserBase.py
import sys
from antlr4 import *
if sys.version_info[1] > 5:
    from typing import TextIO
else:
    from typing.io import TextIO

class LuaParserBase(Parser):

    debug = False

    def __init__(self, input:TokenStream, output:TextIO = sys.stdout):
        super().__init__(input, output)
    
    def IsFunctionCall(self) -> bool:
        la = self._input.LT(1)
        if la.type != self.NAME:
            return False
        la = self._input.LT(2)
        if la.type == self.OP:
            return False
        return True

